packages/bun-usockets/src/context.c

```diff
@@ -394,8 +394,11 @@ struct us_connecting_socket_t *us_socket_context_connect(int ssl, struct us_sock
     if (Bun__addrinfo_get(loop, host, port, &ptr) == 0) {
         // Fast-path: it's already cached.
         // Avoid the connection logic.
-        *is_connecting = 1;
-        return (struct us_connecting_socket_t *) us_socket_context_connect_resolved_dns(context, ptr, options, socket_ext_size);
+
+        // with happy eyeballs we can't use the fast path since we need to initiate multiple connections
```

Comment:
this is not quite accurate

we only should go through the happy eyeballs case when there are multiple results from addrinfo

Reply:
makes sense, I updated the code to use the fast path when there is a cached addrinfo result that has only one entry